source/shared_lib/include/util/callback.h

```diff
@@ -0,0 +1,71 @@
+#pragma once
```

Comment:
This is done in other places of the code with https://github.com/MegaGlest/megaglest-source/blob/develop/source/shared_lib/include/platform/common/simple_threads.h#L71
Although honestly your way seems cleaner to me.